rules/prefer-global-this.js

```diff
@@ -0,0 +1,375 @@
+	/** @param {import('estree').NewExpression} node */
+	NewExpression(node) {
+		handleNodes(context, node.callee);
+	},
+	/** @param {import('estree').ObjectExpression} node */
+	ObjectExpression(node) {
+		for (const property of node.properties) {
+			handleNodes(context, property.value);
+		}
+	},
+	/** @param {import('estree').SequenceExpression} node */
+	SequenceExpression(node) {
+		handleNodes(context, node.expressions);
+	},
+	/** @param {import('estree').TaggedTemplateExpression} node */
+	TaggedTemplateExpression(node) {
+		handleNodes(context, node.tag);
+		handleNodes(context, node.quasi.expressions);
+	},
+	/** @param {import('estree').TemplateLiteral} node */
+	TemplateLiteral(node) {
+		handleNodes(context, node.expressions);
+	},
+	/** @param {import('estree').UnaryExpression} node */
+	UnaryExpression(node) {
+		handleNodes(context, node.argument);
+	},
+	/** @param {import('estree').UpdateExpression} node */
+	UpdateExpression(node) {
+		handleNodes(context, node.argument);
+	},
+
+	// ===== Statements =====
+
+	/** @param {import('estree').ExpressionStatement} node */
+	ExpressionStatement(node) {
+		switch (node.expression.type) {
+			case 'Identifier': {
+				handleNodes(context, node.expression);
+				break;
+			}
+
+			default: {
+				break;
+			}
+		}
+	},
+	/** @param {import('estree').ForStatement} node  */
+	ForStatement(node) {
+		handleNodes(context, [node.init, node.test, node.update]);
+	},
+	/** @param {import('estree').ForInStatement} node  */
+	ForInStatement(node) {
+		handleNodes(context, [node.left, node.right]);
+	},
+	/** @param {import('estree').ForOfStatement} node  */
+	ForOfStatement(node) {
+		handleNodes(context, [node.left, node.right]);
+	},
+	/** @param {import('estree').ReturnStatement} node */
+	ReturnStatement(node) {
+		handleNodes(context, node.argument);
+	},
+	/** @param {import('estree').SwitchStatement} node */
+	SwitchStatement(node) {
+		handleNodes(context, node.discriminant);
+
+		for (const caseNode of node.cases) {
+			handleNodes(context, caseNode.test);
+		}
+	},
+	/** @param {import('estree').WhileStatement} node */
+	WhileStatement(node) {
+		handleNodes(context, node.test);
+	},
+	/** @param {import('estree').DoWhileStatement} node */
+	DoWhileStatement(node) {
+		handleNodes(context, node.test);
+	},
+	/** @param {import('estree').IfStatement} node */
+	IfStatement(node) {
+		handleNodes(context, node.test);
+	},
+	/** @param {import('estree').ThrowStatement} node */
+	ThrowStatement(node) {
+		handleNodes(context, node.argument);
+	},
+	/** @param {import('estree').TryStatement} node */
+	TryStatement(_node) {},
+	/** @param {import('estree').CatchClause} node */
+	CatchClause(_node) {},
+
+	// ===== Declarations =====
+
+	/** @param {import('estree').VariableDeclarator} node */
+	VariableDeclarator(node) {
+		handleNodes(context, node.init);
+	},
+	/** @param {import('estree').AssignmentPattern} node */
+	AssignmentPattern(node) {
+		handleNodes(context, node.right);
+	},
```

Comment:
We can't write rule like this.

Have you tried `GlobalReferenceTracker` metioned in https://github.com/sindresorhus/eslint-plugin-unicorn/pull/2410#issuecomment-2252217275?

If it doesn't fit your needs, we have [`isReferenceIdentifier`](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/77f32e5a6b2df542cf50dfbd371054f2cd8ce2d6/rules/ast/is-reference-identifier.js#L150C10-L150C31) may help too.

Reply:
Okay, I'll take a look at how to use this.

Reply:
The code has been Refactored with GlobalReferenceTracker without change to any test case. 817d8aa36b27588c350ca077664f6afd3d5b7433